Lib/fontTools/merge.py

```diff
@@ -975,20 +1084,43 @@ def merge(self, fontfiles):
 
 		return mega
 
-	def _mergeGlyphOrders(self, glyphOrders):
+	def _mergeGlyphOrders(self, glyphOrders, cffTables):
 		"""Modifies passed-in glyphOrders to reflect new glyph names.
 		Returns glyphOrder for the merged font."""
 		# Simply append font index to the glyph name for now.
 		# TODO Even this simplistic numbering can result in conflicts.
 		# But then again, we have to improve this soon anyway.
+
+		# If we have CFF tables,
+		# rename topDictIndex charStrings
+		for n,cffTable in enumerate(cffTables):
+			td = cffTable.cff.topDictIndex[0]
+			d = {}
+			for i, (glyphName, v) in enumerate(td.CharStrings.charStrings.items()):
+				if glyphName != '.notdef':
+					glyphName += "#" + repr(n)
+				elif n > 0:
+					glyphName += "#" + repr(n)
+				d[glyphName] = v
+			cffTable.cff.topDictIndex[0].CharStrings.charStrings = d
+
 		mega = []
+
+		# If we have CFF tables,
```

Comment:
@BlackFoundry this comment "If we have CFF tables, make sure to keep .notdef..." seems to also apply for non-CFF fonts, so the first part of the comment is misleading

Reply:
True as well (old comment, the code changed after writing it)